File: cs/sample/Sample/ConvertBinaryLinkedList2Int/ConvertBinaryLinkedList2Int.cs
using System;
using System.Collections.Generic;
using Sample.ListNode;

namespace Sample.ConvertBinaryLinkedList2Int{
    public class convertBinaryLinkedList2Int{
        public static int naive(listNode l){
            int result = 0;
            while (l!=null){
                result = result*2+l.val;
                l = l.next;
            }
            return result;
        }
    }
}

File: cpp/example/src/ConvertBinaryLinkedList2Int/convertBinaryLinkedList2Int.cpp
#include "convertBinaryLinkedList2Int.h"

int convertBinaryLinkedList2Int::naive(ListNode* l){
    int result = 0;
    while (l!=nullptr){
        result = result*2+l->val;
        l = l->next;
    }
    return result;
}
